src/traits/election/vrf.rs

```diff
@@ -205,25 +207,44 @@ where
 {
     type PrivateKey = (SIGSCHEME::SigningKey, SIGSCHEME::VerificationKey);
 
-    fn validate(&self, signature: &EncodedSignature, data: &[u8]) -> bool {
-        let x: Result<SIGSCHEME::Signature, _> = bincode_opts().deserialize(&signature.0);
+    fn validate(&self, ver_key: QCVerKey, signature: &EncodedSignature, data: &[u8]) -> bool {
+        let x: Result<<BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature, _> = 
+            bincode_opts().deserialize(&signature.0);
         match x {
             Ok(s) => {
                 // First hash the data into a constant sized digest
-                SIGSCHEME::verify(&(), &self.pk, data, &s).is_ok()
+                // This is the validation for QC partial signature before append().
+                let generic_msg: &GenericArray<u8, U32> = GenericArray::from_slice(data);
+                BLSOverBN254CurveSignatureScheme::verify(
+                    &(),
+                    &ver_key, 
+                    &generic_msg,
+                    &s,
+                ).is_ok() //unwrap();
             }
             Err(_) => false,
         }
     }
 
-    fn sign(private_key: &Self::PrivateKey, data: &[u8]) -> EncodedSignature {
+    fn sign(key_pair: QCKeyPair, 
+            data: &[u8]) -> EncodedSignature {
         // Sign it
-        let signature = SIGSCHEME::sign(&(), &private_key.0, data, &mut rand::thread_rng())
-            .expect("This signature shouldn't be able to fail");
+        let generic_msg = GenericArray::from_slice(data);
+        
+        let individual_signature =
+            BitvectorQuorumCertificate::<BLSOverBN254CurveSignatureScheme>::partial_sign(
+                &(),
+                &generic_msg,
+                key_pair.sign_key_ref(),
+                &mut rand::thread_rng()
+            ).unwrap();
```

Comment:
Can this ever panic? If so, we may need to switch `SignatureKey::sign` to return a `Result<EncodedSignature, SomeError>`. This is in the core validator code and seems like a possibility here, right?

Reply:
This is the signing function, it should not fail in normal cases. I simply change `unwrap()` to `expect("Individual signing should not fail")`, is this okay?